src/test/java/com/newrelic/agent/instrumentation/verify/AfterEvaluationActionTest.java

```diff
@@ -0,0 +1,138 @@
+package com.newrelic.agent.instrumentation.verify;
+
+import org.eclipse.aether.repository.RemoteRepository;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.UnknownProjectException;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.internal.impldep.org.testng.Assert;
+import org.gradle.testfixtures.ProjectBuilder;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.slf4j.Logger;
+
+import java.io.File;
+import java.util.*;
+import java.util.function.Function;
+
+import static com.newrelic.agent.instrumentation.verify.VerificationPlugin.VERIFIER_TASK_NAME;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class AfterEvaluationActionTest {
+
+    private AfterEvaluationAction testClass;
+    private Function<Project, List<RemoteRepository>> getRepositoryFunction;
+
+    @BeforeEach
+    void beforeEach() {
+        MockitoAnnotations.initMocks(this);
+        this.getRepositoryFunction = project -> Collections.emptyList();
+        this.testClass = new AfterEvaluationAction(mockOptions, mockTask, mockLogger, mockDestinationDir, getRepositoryFunction);
+    }
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
```

Comment:
The deep stubs allows much easier mocking of project making chained method calls, to avoid npe.